src/dqe_idx.erl

```diff
@@ -227,6 +228,28 @@ metrics(Collection) ->
 
 %%--------------------------------------------------------------------
 %% @doc
+%% Returns a list of metric path suffixes of `Depth' that are prefixed
+%% by the given probe `Prefix', which can also be empty.
+%%
+%% For example:
+%% metrics(<<"collection">>, <<>>, 1) -> [<<"base">>].
+%% metrics(<<"collection">>, <<"base">>, 1) -> [<<"cpu">>].
+%% metrics(<<"collection">>, <<>>, 2) -> [[<<"base">>.<<"cpu">>]].
+%%
+%% Actual results are binary-encoded in `dproto' encoding.
+%% @end
+%%--------------------------------------------------------------------
+-spec metrics(Collection::collection(),
+              Prefix::metric(),
+              Depth::pos_integer()) ->
+                    {ok, [Metric::metric()]} |
```

Comment:
The results in `metrics/3` are now encoded using `dproto:metric_from_list`, as they are in `metrics/1`.  Removing the examples is the best way forward here.